### ISSUE TITLE
Migration fails

### ISSUE THREAD

**Issue by HectorFGarcia** [realm/realm-java#1605]
I updated Realm to 0.83.1 and now I'm having some migration problems.

On my migration class I have this

if (version == 2){
            Table table = realm.getTable(User.class);
            table.convertColumnToNotNullable(table.getColumnIndex("id"));
            version++;
        }

But when I try to run the app, this happens:

io.realm.exceptions.RealmMigrationNeededException: Field 'id' does support null values in the existing Realm file. Use corresponding boxed type for field 'id' or migrate using io.realm.internal.Table.convertColumnToNotNullable().

And this is the field:

public class Producto extends RealmObject {

    @PrimaryKey
    private long id;

Any help, please?

**Comment by emanuelez** [comment]
What did you model look like in version 1 and version 2?

**Comment by HectorFGarcia** [comment]
I didn't change the field 'id', It's the same in both versions.

**Comment by emanuelez** [comment]
I see. In that case you don't need to migrate the field at all! `long` is a primitive, so it's not nullable by default.

**Comment by HectorFGarcia** [comment]
I tried to migrate because before that I was getting a RealmMigrationNeededException. I changed some fields (Strings) and they work, but with 'id' I'm getting the exception from #1 even if I remove that line.

**Comment by emanuelez** [comment]
So, what changes did you do to your model in total?

**Comment by HectorFGarcia** [comment]
Sorry, I explained myself wrong. I did the changes in the migration class using the method "convertColumnToNullable". I had some exceptions about some Strings and that solved it fine. Then the "id" exception described in #1 showed up. I tried to use "convertColumnToNotNullable" but didn't work.

I didn't touch the model at all, It's the same.

**Comment by emanuelez** [comment]
If you didn't change the model at all and you are migrating from version <= 0.82.2 to 0.83.x then you gave two options:

 1) If you want to keep strings non-nullable then you need to apply the `@Required` annotation.
 2) If you want the String to be nullable then you need to perform a migration such as:

```java
RealmMigration migration = new RealmMigration() {
    @Override
    public long execute(Realm realm, long version) {
        Table table = realm.getTable(Dog.class);
        table.convertColumnToNullable(table.getColumnIndex("name"));
        return 1;
    }
};

RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext())
    .schemaVersion(1)
    .schema(Dog.class)
    .migration(migration)
    .build();
```

Did you read our blog post about it? https://realm.io/news/realm-java-0.83.0/

**Comment by HectorFGarcia** [comment]
I think you aren't understanding me. Let me try again:

1. I updated Realm. I tried to to run the app and I had exceptions about String fields.

2. I read the news post about the new version.

3. I added to my migration class (implements RealmMigration and overritdes excecute) code trying to update from v0 to v1 using the "convertColumnToNullable" method over the String fields. It worked, exceptions related to my String fields stopped.

4. Instead, a new exception showed up: io.realm.exceptions.RealmMigrationNeededException: Field 'id' does support null values in the existing Realm file. Use corresponding boxed type for field 'id' or migrate using io.realm.internal.Table.convertColumnToNotNullable().

5. To fix it, I tried to use the code in #1 but doesn't work. Now I don't know what to do to avoid that exception.
